### ISSUE TITLE
(pipelines): Allow using S3 Event Notifications as pipeline trigger.

### ISSUE THREAD

**Issue by SantiRaposo** [aws/aws-cdk#26894]
### Describe the feature

Currently, the S3 triggers that CDK Pipelines support are described here:
https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_codepipeline_actions.S3Trigger.html

However, a simpler way to enable eventbridge notifications for an S3 bucket exists as a bucket level setting:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/EventBridge.html

However when using `S3Trigger.EVENTS` the generated event pattern is:
```json
{
  "detail-type": ["AWS API Call via CloudTrail"],
  "source": ["aws.s3"],
  "detail": {
    "requestParameters": {
      "bucketName": ["<bucket-name>"],
      "key": ["<buckt-key>"]
    },
    "resources": {
      "ARN": ["<object-arn>"]
    },
    "eventName": ["CompleteMultipartUpload", "CopyObject", "PutObject"]
  }
}
```

Which does not match with the event generated by the S3 integration.

### Use Case

Allow using event-triggered pipes (which AWS recommends https://docs.aws.amazon.com/codepipeline/latest/userguide/update-change-detection.html#update-change-detection-S3-event) without having to provision a CloudTrail trail.

### Proposed Solution

Another enum value that generates an event pattern similar to the following:

```json
{
  "source": [
    "aws.s3"
  ],
  "detailType": [
    "Object Created",
    "Object Restore Completed"
  ],
  "detail": {
    "bucket": {
      "name": "<bucket-name>"
    },
    "object": {
      "key": "<object-key>"
    }
  }
}
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.93.0

### Environment details (OS name and version, etc.)

macOS 13.5.1

**Comment by pahud** [comment]
Makes sense for me. And this is the recommended configuration according to the [document](https://docs.aws.amazon.com/codepipeline/latest/userguide/update-change-detection.html#update-change-detection-overview).

**Comment by tgmedia-nz** [comment]
same here - please prioritise :)
